src/lib/Field.tsx

```diff
@@ -25,11 +27,27 @@ export const Field = <FP extends FieldPath.FieldPath<any, any> = string>({
   const isDirty = useSelector(() => toolkit.isFieldDirty(fieldPath));
   const [isValid, setIsValid] = useState(true);
 
-  const refs = useRef({ validate });
+  const refs = useRef({ validate, promiseCounter: 0 });
   Object.assign(refs.current, { validate });
 
   useEffect(() => {
-    setIsValid(refs.current.validate?.(value) ?? true);
+    const promiseId = ++refs.current.promiseCounter;
```

Comment:
Please use `lodash.uniqueId` > e.g `uniqueId("validateToken")`
also maybe lets change `promiseCounter` to `validateToken`